pypdf/_reader.py

```diff
@@ -1827,6 +1832,13 @@ def decode_permissions(self, permissions_code: int) -> Dict[str, bool]:
             for key, flag in permissions_mapping.items()
         }
 
+    @property
+    def user_access_permissions(self) -> Optional[UserAccessPermissions]:
+        """Get the user access permissions for encrypted documents. Returns None if not encrypted."""
+        if self._encryption is None:
+            return None
+        return UserAccessPermissions(self._encryption.P)
+
```

Comment:
Proposition
```suggestion
    def user_access_permissions(self, other_permissions : Optional[int] = None) -> Optional[UserAccessPermissions]:
        """Get the user access permissions for encrypted documents. 
            Args:
                reserved for Future.should be ignored or set to None.
            Returns:
                Decoded access permissions.None if not encrypted."""
        if self._encryption is None:
            return None
        return UserAccessPermissions(other_permissions if other_permissions is not None else self._encryption.P)

```

Reply:
I am confused which value this might provide over the current situation. The user can still use the `UserAccessPermissions` class directly - is there any need to use the corresponding `PdfReader` for it?

If we are really talking about preparing certificate-based stuff here, then I would still argue that *pypdf* should be able to detect the correct permissions itself and not take them from a parameter to a method on the reader.

Reply:
It's also unclear to me.

I'm sorry @pubpub-zz , but I'll go with the PR of stefan as it is right now. We would then provide a similar interface to PyMuPDF, so it cannot be that bad.

I'm always open to change such decisions, but it might take a long time due to the deprecation policy. However, I have a feeling that the cases you want to prepare pypdf for are super rare and thus likely not relevant for most users. To be honest, I also think that the document permissions are more a "pdf geek" topic than something the broad majority of users care about :sweat_smile: